client/state/site-settings/reducer.js

```diff
@@ -64,7 +66,21 @@ export const items = createReducer( {}, {
 			...state[ siteId ],
 			...settings
 		}
-	} )
+	} ),
+	[ MEDIA_DELETE ]: ( state, { siteId, mediaIds } ) => {
```

Comment:
Hmm, you raise a good point. The fallback exists in case we ever receive site settings before the site itself. But if that were ever to occur, we would still want to monitor for media deletions even if we only knew the site settings. But we still do not need to account for `SITE_RECEIVE` or `SITES_RECEIVE` in the site settings reducer because at that point we can assume that the fallback will no longer be used.

Because of this, I'll plan to drop 15cb3f5 in a rebase.